### ISSUE TITLE
Make fluent accept strings in place of functions

### ISSUE THREAD

**Issue by Farenheith** [codibre/fluent-iterable#14]
Good suggestion from @danstarns
It would be nice if all or, at least, some operations of fluent also accepts string values where today it is accepted only functions.

Example:

```ts
// this line
fluent(it).map((x) => x.field);
// Could also be:
fluent(it).map('field');
```

### Tip:
Maybe a good strategy for this is to start improving the predicate types, like this:
```ts
export interface FunctionPredicate<T> {
  (item: T): any;
}
export type Predicate<T> = FunctionPredicate<T> | keyof T
...
export type AnyMapper<T> = (value: T) => any | keyof T;
```
The recipes must be checked too to make sure that all the mappers and predicates used are typed. This change may break all places where a predicate is used, so, it will be easier to track down where it needs to be changed.
This is a big big change but still must be implemented as **minor**, and will require a new unit test for each method that receives functions as parameters.

### TIP 2
Analyze if [object-path](https://www.npmjs.com/package/object-path) can be used with this feature for a more robust approach

**Comment by danstarns** [comment]
Using something like [object-path](https://www.npmjs.com/package/object-path) to access nested properties could be usefull too. 

```ts
// this line
fluent(it).map((x) => x.a.b.c.field);
// Could also be:
fluent(it).map('x.a.b.c.field');
```

**Comment by Farenheith** [comment]
@danstarns I thought about something like this:
```ts
fluent(it).map(f('x').a.b.c.field)
```
The good thing about this that I can use **Proxy** to control the expected nested properties and could provide the user with IntelliSense through a good typing.
What do you think? Do you think it'd look too weird?

**Comment by danstarns** [comment]
@Farenheith would `f('x')` be a function as part of the package? I don't think I understand what you are trying to express.

**Comment by Farenheith** [comment]
@danstarns 
The object-path is a good package, but it relies greatly on the user typing. I thought I could create a function that returns an "object navigator". let's call this function **f**.

```ts
const navigator = f<MyType>('field');
```
With this, I could chain properties navigation through the object:
```ts
navigator.field2 // represents obj.field.field2, may be converted to x => x.field.field2
navigator.field2.field3 // represents obj.field.field2, may be converted to x => x.field.field2.field3
```

I'm not sure yet but I think TypeScript can infer **MyType** in this usage:
```ts
// it is a Iterable<MyType>
fluent(it).map(f('x').a.b.c.field)
```

My concern about this idea is that the object-path syntax is more intuitive at first view, though you can't have auto-complete with it, so, I don't know if it is worth it to invest time in this **f.** solution.

There's nothing that stops me of using **object-path** at first, as well and I can evolve this idea in another issue, or even in another project

**Comment by danstarns** [comment]
@Farenheith 

Thanks for the detailed reply, I see what you're trying to say. Have you thought about names for this function? I see **Navigator** was used.

**Comment by Farenheith** [comment]
@danstarns 
I didn't think yet.
I think it's a good idea for a new package, I'll look into this with more attention later, but for now I'll stick with the object-path suggestion :)

Thanks for yours tips!

**Comment by Farenheith** [comment]
@danstarns I took some time thinking of it and maybe **json-navigator** is a good name. Also, I wrote a README [here](https://github.com/Codibre/json-navigator/blob/master/README.md) of how I think it could work and why it may be useful. If you have some time I'd appreciate it if you take a look and give me an honest opinion if a package like this is worth it to be implementeted.
I also did some tests in the index.ts just to make sure the Typing control I thought is possible.
